stripe.go

```diff
@@ -74,6 +74,54 @@ var Key string
 // Public types
 //
 
+// APIResponse encapsulates some common features of a response from the
+// Stripe API.
+type APIResponse struct {
+	// Header contain a map of all HTTP header keys to values. Its behavior and
+	// caveats are identical to that of http.Header.
+	Header http.Header
+
+	// IdempotencyKey contains the idempotency key used with this request.
+	// Idempotency keys are a Stripe-specific concept that helps guarantee that
+	// requests that fail and need to be retried are not duplicated.
+	IdempotencyKey string
+
+	// RawJSON contains the response body as raw bytes.
+	RawJSON []byte
+
+	// RequestID contains a string that uniquely identifies the Stripe request.
+	// Used for debugging or support purposes.
+	RequestID string
+
+	// Status is a status code and message. e.g. "200 OK"
+	Status string
+
+	// StatusCode is a status code as integer. e.g. 200
+	StatusCode int
+}
+
+func newAPIResponse(res *http.Response, resBody []byte) *APIResponse {
+	return &APIResponse{
+		Header:         res.Header,
+		IdempotencyKey: res.Header.Get("Idempotency-Key"),
+		RawJSON:        resBody,
+		RequestID:      res.Header.Get("Request-Id"),
```

Comment:
Minor but are those headers access safe? I'm asking because we had some weird things with HTTP2 last year and the case changing

Reply:
Yep, `Header.Get` provides totally case-insensitive access. See docs here:

https://godoc.org/net/http#Header.Get